### PR TITLE
Feature/328 call activity labels

### LINKED ISSUE #328: Call Activity Flow Name
When a start event is configured for a call activity flow, set the flow element label as the selected start event name.

### LINKED ISSUE #334: SequenceFlow between Gateway and CallActivity
Currently if the sequence flow outcomes a Gateway and incomes a CallActivity does not have the Start Event property.

image

### PR DESCRIPTION
**Pull request by chipit24**
Fixes #328, fixes #334.

How to test:
* Drag an exclusive gateway and a call activity onto the grid and connect them with a sequence flow.
* Select the sequence flow and add an expression; the expression should appear above the sequence flow.
* Select the call activity and in the inspector panel, select a process under the `Process` dropdown.
* Select the sequence flow again and this time, select a start event under the `Call Activity Start Event` dropdown in the inspector panel. The start event name should appear above the sequence flow.

### CODE REVIEW

File: src/components/inspectors/InspectorPanel.vue [hunk 1]
```diff
@@ -76,24 +76,38 @@ export default {
...
+        const expressionConfig = {
+          component: 'FormInput',
+          config: {
+            label: 'Expression',
+            helper: 'Enter the expression that describes the workflow condition',
+            name: 'conditionExpression',
+          },
+        };
+
+        inspectorConfig[0].items[1].items.push(expressionConfig);
```
Comment (tuantruong625) [inline src/components/inspectors/InspectorPanel.vue]: For readability, do you think extracting  **inspectorConfig[0].items[1].items** to `const sequenceFlowConfig = inspectorConfig[0].items[1].items`, would be better ?

File: src/components/inspectors/InspectorPanel.vue [hunk 2]
```diff
@@ -76,24 +76,38 @@ export default {
...
+            name: 'startEvent',
+            targetCallActivity: definition.targetRef,
+            helper: definition.targetRef.calledElement
+              ? ''
+              : 'Please select a valid process on the connected call activity.',
+          },
+        };
 
-        return sequenceCallActivity;
+        inspectorConfig[0].items[1].items.push(startEventConfig);
```
Comment (tuantruong625) [inline src/components/inspectors/InspectorPanel.vue]: Same as comment posted above.

File: src/components/nodes/callActivity/CallActivityFormSelect.vue [hunk 3]
```diff
@@ -9,48 +9,53 @@
...
-            subprocessList.push(event.ownerProcessId);
-            list.push(this.toDropdownFormat(process, event));
-          }
+
+      store.getters.globalProcesses.forEach(process => {
+        uniqBy(process.events, 'ownerProcessId').forEach(event => {
+          list.push(this.toDropdownFormat(process, event));
         });
+
+        // const subprocessList = [];
```
Comment (tuantruong625) [inline src/components/nodes/callActivity/CallActivityFormSelect.vue]: Is this code needed for anything ?

File: src/components/nodes/callActivity/CallActivityFormSelect.vue [hunk 4]
```diff
@@ -9,48 +9,53 @@
...
     containsMultipleProcesses(process) {
-      const subprocessList = [];
-      process.events.forEach((event) => {
-        if (subprocessList.indexOf(event.ownerProcessId)===-1) {
-          subprocessList.push(event.ownerProcessId);
-        }
-      });
-      return subprocessList.length > 1;
+      return uniqBy(process.events, 'ownerProcessId').length > 1;
+      // const subprocessList = [];
```
Comment (tuantruong625) [inline src/components/nodes/callActivity/CallActivityFormSelect.vue]: Another comment

File: src/components/nodes/callActivity/CallActivityFormSelect.vue [hunk 5]
```diff
@@ -9,48 +9,53 @@
...
+      //     subprocessList.push(event.ownerProcessId);
+      //   }
+      // });
+      // return subprocessList.length > 1;
     },
     toDropdownFormat(process, event) {
       return {
-        value: event.ownerProcessId + '-' + process.id,
-        content: (process.name || process.id) + ( this.containsMultipleProcesses(process) ? '(' + event.ownerProcessName + ')' : ''),
+        value: `${event.ownerProcessId}-${process.id}`,
```
Comment (tuantruong625) [inline src/components/nodes/callActivity/CallActivityFormSelect.vue]: Nice, refactor.

File: src/components/nodes/callActivity/index.js [hunk 6]
```diff
@@ -68,10 +69,18 @@ export default {
...
+                label: 'Name',
+                helper: 'The Name of the Call Activity',
+                name: 'name',
+              },
+            },
             {
               component: CallActivityFormSelect,
               config: {
-                label: 'Processe',
+                label: 'Process',
```
Comment (tuantruong625) [inline src/components/nodes/callActivity/index.js]: Good, catch.

File: src/components/nodes/callActivity/CallActivityFormSelect.vue (hunk 3 above)
Comment (chipit24) [reply src/components/nodes/callActivity/CallActivityFormSelect.vue]: Nope, forgot to remove it!

File: src/components/nodes/callActivity/CallActivityFormSelect.vue [hunk 7]
```diff
@@ -9,48 +9,37 @@
...
 <script>
 import store from '@/store';
+import uniqBy from 'lodash/uniqBy';
 
 export default {
   computed: {
     dropdownList() {
       return this.processList.length > 0
-        ? [{ value: '', content: 'Select Active Process' }, ...this.processList]
+        ? [{ value: '', content: '[Select Active Process]' }, ...this.processList]
```
Comment (caleeli) [inline src/components/nodes/callActivity/CallActivityFormSelect.vue]: '[Select Active Process]' is not translated

**Pull request merged**